### ISSUE TITLE
403's when a long @Async process runs: vault should not publish a ThreadPoolTaskScheduler bean on Spring's context

### ISSUE THREAD

**Issue by hjohn** [spring-projects/spring-vault#623]
The fix in https://github.com/spring-projects/spring-vault/issues/308 is still interfering with other Spring code using vault.  Vault still publishes a Bean for everyone to use, as Spring will match the bean by type if it is the only one.

This bean should not be public at all as it is (apparently) vital for Spring Vault's functioning.  If an unrelated method uses this bean by accident (which is easy to do as it used by `@Async`) and runs a long process (blocking vault code as the executor only has 1 thread) it can interfere with Vault's refresh mechanisms.  This interference will show up as 403's on all secrets being requested.

I would suggest NOT publishing this bean in Spring's context.  From what I can quickly glance from the code (with `grep`), it doesn't need to be a bean at all (the name `vaultThreadPoolTaskScheduler` is not used anywhere apart from declaration).  Make it a static private field in  `AbstractVaultConfiguration` and have the code calling `threadPoolTaskScheduler` just use that.

**Comment by mp911de** [comment]
Thanks for report. In Spring Vault we exposed a scheduler bean directly while in Spring Cloud Vault we added a wrapper (`TaskSchedulerWrapper`) to not interfere with these beans. We could wrap the bean and make the retrieval method `getVaultThreadPoolTaskScheduler` overridable so you get more control over the desired bean to look up. Likely, we could introduce that change for 2.4.1 since 2.4.0 is out for about a month.
